web/handlers_test.go

```diff
@@ -359,6 +359,44 @@ func TestHandlerServeCSPHeader(t *testing.T) {
 		// TODO: See above.
 		// assert.Contains(t, response.Header()["Content-Security-Policy"], "frame-ancestors 'self'; script-src 'self' cdn.rudderlabs.com 'sha256-tPOjw+tkVs9axL78ZwGtYl975dtyPHB6LYKAO2R3gR4='", "csp header incorrectly changed after subpath changed")
 	})
+
+	t.Run("static, development mode enabled", func(t *testing.T) {
+		th := SetupWithStoreMock(t)
+		defer th.TearDown()
+
+		mockStore := th.App.Srv().Store.(*mocks.Store)
+		mockPostStore := mocks.PostStore{}
+		mockPostStore.On("GetMaxPostSize").Return(65535, nil)
+		mockSystemStore := mocks.SystemStore{}
+		mockSystemStore.On("GetByName", "UpgradedFromTE").Return(&model.System{Name: "UpgradedFromTE", Value: "false"}, nil)
+		mockSystemStore.On("GetByName", "InstallationDate").Return(&model.System{Name: "InstallationDate", Value: "10"}, nil)
+		mockSystemStore.On("GetByName", "FirstServerRunTimestamp").Return(&model.System{Name: "FirstServerRunTimestamp", Value: "10"}, nil)
+
+		mockStore.On("Post").Return(&mockPostStore)
+		mockStore.On("System").Return(&mockSystemStore)
```

Comment:
NB: WIthout mocking the store(s) here, the following fatal error occurred during unit testing:

```
fatal error: sync: Unlock of unlocked RWMutex
```

That seemed to be something to do with [retrieval of `Store.Post().GetMaxPostSize()` here](https://github.com/jayaddison-collabora/mattermost-server/blob/597c9dab3410fff183dd503e9ff94e798d41d850/app/post.go#L1428).

Reply:
There seems to be some cyclic dependency that causes a panic there. I am wondering if it would be simpler (and less verbose) to go with the standard server bootup through `Setup(t).InitBasic()` given we don't really need mocks to test this feature. Then we can just do an `http.Get()` call on `apiClient.URL` and check the response for the expected headers. Also, it would be good to check for both config values (true and false).

Reply:
Thanks @streamer45 - yep, that sounds good, I'll give that a try :+1: